src/tls/pem.h

```diff
@@ -44,6 +45,14 @@ namespace tls
       return reinterpret_cast<const uint8_t*>(s.data());
     }
 
+    std::vector<uint8_t> raw() const
```

Comment:
I know we have #601 to resolve this in a separate PR, but if we can fix the null-termination now (returning `s.size() + 1` from size, stripping the existing null if necessary in constructor) then we can avoid this `raw()` method and associated copy.